scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -935,8 +935,8 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    * Window values with the given function.
    * @group window
    */
-  def withWindowFn[W <: BoundedWindow](
-    fn: WindowFn[AnyRef, W],
+  def withWindowFn[A <: AnyRef, W <: BoundedWindow](
+    fn: WindowFn[A, W],
```

Comment:
Hi @mkuthan on a second thought, since `WindowFn` is applied on a `SCollection[T]` we should apply here a generic lower bound constraint instead. This way no need for a second type parameter.

```suggestion
    fn: WindowFn[_ >: T, W],
```

Reply:
Indeed! Even safer!

Reply:
Ahh unfortunately `2.12` doesn't like this ... I think we will need to go with smth `_ <: Any` (thinking of `AnyVal`'s)

Reply:
Took the liberty of going with this! Hope you don't mind @mkuthan and thanks a lot for spotting this!

Reply:
Another alternative would be to have diff constraints for 2.12/2.13 🤔 not sure if I want to go that route ... but we would end up with a better 2.13.